### DISCUSSION TITLE
AgentGroupChat vs. Process Framework

### DISCUSSION THREAD

**Discussion by deepinderdeol** [microsoft/semantic-kernel#9099] [Q&A]
Multiple custom agents each with their plugins and instructions can be added to a AgentGroupChat to fulfill a task - so can be used for task automation.
Looks like the Process-Framework can also be used to do something similar.

Can you please provide some guidance on where one option is better than the other OR maybe these are to be used together - if so then how?

**Comment by alliscode** [comment]
Hi @deepinderdeol, thanks for the question. The AgentGroupChat is great when you want to allow agents to participate with each other in a purely conversational (chat-based) way. This can be very powerful for certain situations but can also be challenging to control. The Process framework on the other hand provides a way to structure the flow of information in a controlled event-driven way that may use chat or may be more artifact-driven. You can read more about the process framework here: https://learn.microsoft.com/en-us/semantic-kernel/frameworks/process/process-framework
